errors.md

```diff
@@ -0,0 +1,134 @@
+
+1. ShadowRoot's `Error.stack`:
+```
+foo@http://127.0.0.1:8081/y.js:2:5
+bar@http://127.0.0.1:8081/y.js:6:12
+captureStack@http://127.0.0.1:8081/x.js:9:9
+```
+
+_Note: Only the 3 stack frames must be exposed on this error because they correspond to code evaluated inside the ShadowRealm associated to the error instance._
+
+2. Main page's `Error.stack`:
+```
+foo@http://127.0.0.1:8081/y.js:2:5
+bar@http://127.0.0.1:8081/y.js:6:12
+captureStack@http://127.0.0.1:8081/x.js:9:9
+runInSandbox@http://127.0.0.1:8081/demo.html:6:17
+@http://127.0.0.1:8081/demo.html:8:1
+```
+
+_Note: All 5 stack frames must be exposed on this error regardless of their location and associated Realms, this is analogous to what happen with same domain iframes today._
+
+In the example above, we have 2 modules, `x.js` and `y.js`:
+
+```js
+import { foo, bar } from './y.js';
+
+export function letItThrow() {
+    return foo();
+}
+
+export function captureStack() {
+    try {
+        bar();
+    } catch (e) {
+        return e.stack;
+    }
+    return '';
+}
+```
+
+```js
+export function foo() {
+    return null.x;
+}
+
+export function bar() {
+    return foo();
+}
+```
+
+The incubator realm running (demo.html) can do the following:
+
+```js
+const s = new ShadowRealm();
+const letItThrow = await s.importValue('./x.js', 'letItThrow');
+const captureStack = await s.importValue('./x.js', 'captureStack');
+function runInSandbox() {
+    letItThrow();
+    // or captureStack();
+}
+runInSandbox();
+```
+
+#### When not to censor Errors?
+
+Since ShadowRealm provides integrity guarantees, errors instances belonging to a ShadowRealm are only accessible inside the ShadowRealm (thanks to the callable boundary), this censoring process only affect errors inside a ShadowRealm, developers, and dev-tools observing errors at the top level (main window), should still be able to observe the full stack information. This covers tools to collect metrics about errors, and any other mechanism running on the main program.
+
+### Example
+
+Now, let's assume that we have 3 programs:
+
+ - Main Program (running on top level)
+   - Program A (running inside a ShadowRealm created by the Main Program)
+     - Program B (running inside a ShadowRealm created by Program A)
+
+The same mechanism must be applied if the error was generated by a child ShadowRealm B (Program B), which was created by Program A. Let's explore the different scenarios:
+
+1. A try/catch in Program B catches the error, it will qualify to be censored, only revealing stack frames associated to the ShadowRealm where Program A runs, and none from Program B or the Main Program.
```

Comment:
This is a typo, right?

```suggestion
1. A try/catch in Program B catches the error, it will qualify to be censored, only revealing stack frames associated to the ShadowRealm where Program B runs, and none from Program A or the Main Program.
```

Reply:
Yes, a typo. good catch @bathos